library/core/resource-loader/src/testmod/java/org/quiltmc/qsl/resource/loader/test/BuiltinResourcePackTestMod.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.resource.loader.test;
+
+import static org.quiltmc.qsl.resource.loader.test.ResourceLoaderTestMod.id;
+
+import net.fabricmc.api.ModInitializer;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.quiltmc.qsl.resource.loader.api.ResourceLoader;
+import org.quiltmc.qsl.resource.loader.api.ResourcePackActivationType;
+
+public class BuiltinResourcePackTestMod implements ModInitializer {
+	private static final Logger LOGGER = LogManager.getLogger();
+
+	@Override
+	public void onInitialize() {
+		if (!ResourceLoader.registerBuiltinResourcePack(id("test"), ResourcePackActivationType.DEFAULT_ENABLED)) {
+			LOGGER.warn("Could not register built-in resource pack.");
```

Comment:
this should probably fail hard so it's easily caught when someone is naively running all the test mods at once

Reply:
Throwing a RuntimeException now.